jdbcmapper/src/main/java/com/moparisthebest/jdbc/codegen/JdbcMapperProcessor.java

```diff
@@ -171,6 +172,8 @@ public boolean process(final Set<? extends TypeElement> annotations, final Round
 					final JdbcMapper.Mapper mapper = genClass.getAnnotation(JdbcMapper.Mapper.class);
 					final JdbcMapper.DatabaseType databaseType;
 					final String arrayNumberTypeName, arrayStringTypeName;
+					final boolean generateAsSpringBean = mapper.generateAsSpringBean().combine(false);
```

Comment:
Need to .combine(annotation processor setting).combine(repository is on classpath) or similar

Oh, and correct me if I'm wrong, but I think since this would use the default constructor, and there is only a default constructor if jndiName is set, you should check and make it a compile-time error if generateAsSpringBean is true and jndiName is ""?